sip_assembly/routines_helpers.py

```diff
@@ -68,13 +68,11 @@ def create_rights_csv(bag_path, rights_statements):
     filepath = os.path.join(bag_path, 'data', 'metadata', 'rights.csv')
     for rights_statement in rights_statements:
         csvwriter = setup_csv_file(filepath)
-        for (dirpath, dirnames, filenames) in os.walk(
-                os.path.join(bag_path, 'data', 'objects')):
-            write_rights_row(
-                dirpath.split(bag_path)[1],
-                filenames,
-                rights_statement,
-                csvwriter)
+        for (dirpath, dirnames, filenames) in os.walk(os.path.join(bag_path, 'data', 'objects')):
+            for file in filenames:
+                rights_rows = get_rights_rows(dirpath.split(bag_path)[1], file, rights_statement)
```

Comment:
Worth looking at this first argument - are we just passing "objects" because if so it might be more transparent to just pass that string rather than doing all this splitting and index referral business.